homeassistant/components/here_travel_time/sensor.py

```diff
@@ -0,0 +1,285 @@
+    """Take a string like 08:00:00 and convert it to a unix timestamp."""
+    combined = datetime.combine(
+        dt_util.start_of_local_day(), dt_util.parse_time(timestr)
+    )
+    if combined < datetime.now():
+        combined = combined + timedelta(days=1)
+    return dt_util.as_timestamp(combined)
+
+
+def setup_platform(hass, config, add_entities_callback, discovery_info=None):
+    """Set up the HERE travel time platform."""
+    def run_setup(event):
+        """
+        Delay the setup until Home Assistant is fully initialized.
+
+        This allows any entities to be created already
+        """
+        hass.data.setdefault(DATA_KEY, [])
+
+        travel_mode = config.get(CONF_MODE)
+        traffic_mode = config.get(CONF_TRAFFIC_MODE)
+        route_mode = config.get(CONF_ROUTE_MODE)
+
+        name = config.get(CONF_NAME, DEFAULT_NAME)
+        app_id = config.get(CONF_APP_ID)
+        app_code = config.get(CONF_APP_CODE)
+        origin = config.get(CONF_ORIGIN)
+        destination = config.get(CONF_DESTINATION)
+
+        sensor = HERETravelTimeSensor(
+            hass,
+            name,
+            app_id,
+            app_code,
+            origin,
+            destination,
+            travel_mode,
+            traffic_mode,
+            route_mode,
+        )
+        hass.data[DATA_KEY].append(sensor)
+
+        if sensor.valid_api_connection:
+            add_entities_callback([sensor])
+
+    # Wait until start event is sent to load this component.
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, run_setup)
+
+
+class HERETravelTimeSensor(Entity):
+    """Representation of a HERE travel time sensor."""
+
+    def __init__(
+            self,
+            hass,
+            name,
+            app_id,
+            app_code,
+            origin,
+            destination,
+            travel_mode,
+            traffic_mode,
+            route_mode,
+    ):
+        """Initialize the sensor."""
+        import herepy
+
+        self._hass = hass
+        self._name = name
+        self._travel_mode = travel_mode
+        self._traffic_mode = traffic_mode
+        self._route_mode = route_mode
+        self._unit_of_measurement = 'min'
+        self._response = None
+        self.valid_api_connection = True
+
+        # Check if location is a trackable entity
+        if origin.split('.', 1)[0] in TRACKABLE_DOMAINS:
+            self._origin_entity_id = origin
+        else:
+            self._origin = origin
+
+        if destination.split('.', 1)[0] in TRACKABLE_DOMAINS:
+            self._destination_entity_id = destination
+        else:
+            self._destination = destination
+
+        self._client = herepy.RoutingApi(app_id, app_code)
+        self.update()
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self._response is None:
+            return None
+
+        # pylint: disable=E1101
+        _summary = self._response.response['route'][0]['summary']
```

Comment:
I see you di dthis below too. Instead of duplicaing code for a weird data structure make a wrapper data structure that noirmalizes this so we can just say 


```
hereRoute = HereRoute(response)
_summary = hereRoute.summary()
```

or something similar

Reply:
Makes sense. I could also introduce this datastructure as a PR in the underlying herepy lib. What do you think? Where is the best place?

Reply:
Here would be the best place. It's becoming more common to have a data structure in the itegrations

Reply:
I implemented the same structure as the waze_travel_time component